### ISSUE TITLE
Testing file uploads on master; consumed_uploaded_entries returns empty list.

### ISSUE THREAD

**Issue by richardkmichael** [phoenixframework/phoenix_live_view#1210]
I wanted to build a LiveView page with uploads this week and noticed the uploads merge to master -- thank you!  I thought I'd give it a try before it goes out the door in 0.15 (maybe?).

To begin with a very basic setup, uploading only one file, I've tried to follow the instructions in the new [uploads guide](https://github.com/phoenixframework/phoenix_live_view/blob/master/guides/server/uploads.md).

The `consume_uploaded_entries` call returns an empty list.  I am unsure where to look next to sort it out.  Apologies if I've missed something, I guess it's in the naming somewhere.  I've tried changing the name of the assign and allowed field name.

Here is the setup and LiveView code.

`phoenix_live_view` from `master@b9ce330`:

```elixir
# mix.exs
defp deps do
  [
    # ...deps
    {:phoenix_live_view, override: true, github: "phoenixframework/phoenix_live_view", branch: "master"}
  ]
end
```

Verify with `mix deps` (excerpted):

```
* phoenix 1.5.6 (Hex package) (mix)
  locked at 1.5.6 (phoenix) 0dc4d39a
  ok
...
* phoenix_live_view 0.15.0-dev (https://github.com/phoenixframework/phoenix_live_view.git - origin/master) (mix)
  locked at b9ce330 (branch: master)
  ok
```

Elixir:

```
Erlang/OTP 23 [erts-11.1.1] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:1] [hipe] [dtrace]
Elixir 1.11.1 (compiled with Erlang/OTP 23)
```

The form renders and I can add a file.  When the form is submitted, `uploaded_files |> IO.inspect`
is an empty list, and the callback is never run.

The live view and HTML:

```elixir
defmodule GuideSearchUiWeb.PageLive do
  use GuideSearchUiWeb, :live_view

  @impl true
  def mount(_params, _session, socket) do
    {:ok, socket
          |> assign(min_length: nil, intervals: nil, uploaded_files: [])
          |> allow_upload(:haystack, accept: ~w(.fa), max_entries: 1)}  # I added a MIME type for `.fa`.
  end

  @impl true
  def handle_event("search", _params socket) do
    uploaded_files =
      consume_uploaded_entries(socket, :haystack, fn %{path: path}, _entry ->
        path |> IO.inspect
        dest = Path.join("priv/static/uploads", Path.basename(path))
        File.cp!(path, dest)
        Routes.static_path(socket, "/uploads/#{Path.basename(dest)}")
      end)

    uploaded_files |> IO.inspect

    {:noreply, socket |> update(:uploaded_files, &(&1 ++ uploaded_files))}
  end
```

```html
  <form phx-submit="search">
    <div>
      <%= live_file_input @uploads.haystack %>
    </div>

    <input type="number" name="min_length" value="<%= @min_length %>" step=1000 />

    <button type="submit" phx-disable-with="Searching...">Find intervals</button>
  </form>

  <!-- More snipped.. -->
```

**Comment by mcrumm** [comment]
Hi @richardkmichael - thanks for trying out uploads! Can you add `phx-change="validate"` to your `<form>` tag and add this callback to your LiveView: 

```elixir
  @impl true
  def handle_event("validate", _params, socket) do
    {:noreply, socket}
  end
```

The change event can be a no-op, but without the callback I don't think the uploads are being pre-flighted. 

I published an example app you can check against, too- https://github.com/mcrumm/live_upload_example

**Comment by richardkmichael** [comment]
@mcrumm Thanks for the quick reply!  Sorry I couldn't get to this over the weekend.

I added the code you suggested (`phx-change` and the no-op handler), and found that the page would not load; indefinite spin.  I noticed your example form also adds `phx-auto-recover="ignore"`, so I added that and the page then loads.  I have not needed the `phx-auto-recover` attribute before; I see it's related to LiveView crashes/disconnects.  I'm uncertain why it's required.

Also, I had to upgrade to the tip of the `v1.5` branch (I noticed your example also uses the latest 1.5).  Will uploads depend on Phoenix 1.5.7 ?

If I use released 1.5.6:

```
* phoenix 1.5.6 (Hex package) (mix)
  locked at 1.5.6 (phoenix) 0dc4d39a
  ok
```

This error results:

```
[info] CONNECTED TO Phoenix.LiveView.Socket in 37µs
  Transport: :websocket
  Serializer: Phoenix.Socket.V2.JSONSerializer
  Parameters: %{"_csrf_token" => "KgsKC38wBGpwXDkWQWYEBkQGUkgLPyYAkyeM2AO59oiP02Jsrj188YgR", "_mounts" => "0", "_track_static" => %{"0" => "http://localhost:4000/css/app.css", "1" => "http://localhost:4000/js/app.js"}, "vsn" => "2.0.0"}
[info] JOINED lvu:0 in 624µs
  Parameters: %{"token" => "SFMyNTY.g2gDaAJhBHQAAAACZAADcGlkWGQADW5vbm9kZUBub2hvc3QAAAZzAAAAAAAAAABkAANyZWZoAm0AAAAUcGh4LUZrWGc5bkduZjBpLWhEcUhtAAAAATBuBgAb4KatdQFiAAFRgA.ytitVp7IwSGOpIJnJmY9gMbOcFHKkXEKC3IgdaGB04U"}
[error] GenServer #PID<0.1654.0> terminating
** (FunctionClauseError) no function clause matching in Phoenix.LiveView.UploadChannel.handle_in/3
    (phoenix_live_view 0.15.0-dev) lib/phoenix_live_view/upload_channel.ex:54: Phoenix.LiveView.UploadChannel.handle_in("chunk", %{}, %Phoenix.Socket{assigns: %{chunk_timeout: 10000, chunk_timer: nil, done?: false, handle: #PID<0.1655.0>, live_view_pid: #PID<0.1651.0>, max_file_size: 194, path: "/var/folders/nv/_g8rcnr13ys71f0wl8xqf4780000gn/T//plug-1604/live_view_upload-1604936196-394769126845464-5", uploaded_size: 0}, channel: Phoenix.LiveView.UploadChannel, channel_pid: #PID<0.1654.0>, endpoint: GuideSearchUiWeb.Endpoint, handler: Phoenix.LiveView.Socket, id: nil, join_ref: "9", joined: true, private: %{connect_info: %{session: %{"_csrf_token" => "AroFMqK_I3PFqTNu6lcp3fAR"}}, log_handle_in: false, log_join: :info}, pubsub_server: GuideSearchUi.PubSub, ref: "10", serializer: Phoenix.Socket.V2.JSONSerializer, topic: "lvu:0", transport: :websocket, transport_pid: #PID<0.1645.0>})
    (phoenix 1.5.6) lib/phoenix/channel/server.ex:316: Phoenix.Channel.Server.handle_info/2
    (stdlib 3.13.2) gen_server.erl:680: :gen_server.try_dispatch/4
    (stdlib 3.13.2) gen_server.erl:756: :gen_server.handle_msg/6
    (stdlib 3.13.2) proc_lib.erl:226: :proc_lib.init_p_do_apply/3
Last message: %Phoenix.Socket.Message{event: "chunk", join_ref: "9", payload: %{}, ref: "10", topic: "lvu:0"}
State: %Phoenix.Socket{assigns: %{chunk_timeout: 10000, chunk_timer: nil, done?: false, handle: #PID<0.1655.0>, live_view_pid: #PID<0.1651.0>, max_file_size: 194, path: "/var/folders/nv/_g8rcnr13ys71f0wl8xqf4780000gn/T//plug-1604/live_view_upload-1604936196-394769126845464-5", uploaded_size: 0}, channel: Phoenix.LiveView.UploadChannel, channel_pid: #PID<0.1654.0>, endpoint: GuideSearchUiWeb.Endpoint, handler: Phoenix.LiveView.Socket, id: nil, join_ref: "9", joined: true, private: %{connect_info: %{session: %{"_csrf_token" => "AroFMqK_I3PFqTNu6lcp3fAR"}}, log_handle_in: false, log_join: :info}, pubsub_server: GuideSearchUi.PubSub, ref: nil, serializer: Phoenix.Socket.V2.JSONSerializer, topic: "lvu:0", transport: :websocket, transport_pid: #PID<0.1645.0>}
[error] GenServer #PID<0.1656.0> terminating
** (stop) exited in: GenServer.call(#PID<0.1651.0>, {:phoenix, :register_entry_upload, %{channel_pid: #PID<0.1656.0>, entry_ref: "0", ref: "phx-FkXg9nGnf0i-hDqH"}}, 5000)
    ** (EXIT) no process: the process is not alive or there's no process currently associated with the given name, possibly because its application isn't started
    (elixir 1.11.1) lib/gen_server.ex:1027: GenServer.call/3
    (phoenix_live_view 0.15.0-dev) lib/phoenix_live_view/upload_channel.ex:28: Phoenix.LiveView.UploadChannel.join/3
    (phoenix 1.5.6) lib/phoenix/channel/server.ex:377: Phoenix.Channel.Server.channel_join/4
    (phoenix 1.5.6) lib/phoenix/channel/server.ex:299: Phoenix.Channel.Server.handle_info/2
    (stdlib 3.13.2) gen_server.erl:680: :gen_server.try_dispatch/4
    (stdlib 3.13.2) gen_server.erl:756: :gen_server.handle_msg/6
    (stdlib 3.13.2) proc_lib.erl:226: :proc_lib.init_p_do_apply/3
Last message: {Phoenix.Channel, %{"token" => "SFMyNTY.g2gDaAJhBHQAAAACZAADcGlkWGQADW5vbm9kZUBub2hvc3QAAAZzAAAAAAAAAABkAANyZWZoAm0AAAAUcGh4LUZrWGc5bkduZjBpLWhEcUhtAAAAATBuBgAb4KatdQFiAAFRgA.ytitVp7IwSGOpIJnJmY9gMbOcFHKkXEKC3IgdaGB04U"}, {#PID<0.1645.0>, #Reference<0.2106799845.3591897094.137426>}, %Phoenix.Socket{assigns: %{}, channel: Phoenix.LiveView.UploadChannel, channel_pid: nil, endpoint: GuideSearchUiWeb.Endpoint, handler: Phoenix.LiveView.Socket, id: nil, join_ref: "11", joined: false, private: %{connect_info: %{session: %{"_csrf_token" => "AroFMqK_I3PFqTNu6lcp3fAR"}}}, pubsub_server: GuideSearchUi.PubSub, ref: nil, serializer: Phoenix.Socket.V2.JSONSerializer, topic: "lvu:0", transport: :websocket, transport_pid: #PID<0.1645.0>}}
State: #Reference<0.2106799845.3591897094.137428>
[error] exited in: GenServer.call(#PID<0.1651.0>, {:phoenix, :register_entry_upload, %{channel_pid: #PID<0.1656.0>, entry_ref: "0", ref: "phx-FkXg9nGnf0i-hDqH"}}, 5000)
    ** (EXIT) no process: the process is not alive or there's no process currently associated with the given name, possibly because its application isn't started
[error] GenServer #PID<0.1657.0> terminating
** (KeyError) key "phx-FkXg9nGnf0i-hDqH" not found in: %{"phx-FkXg9842voAwRDqn" => :haystack}
    :erlang.map_get("phx-FkXg9nGnf0i-hDqH", %{"phx-FkXg9842voAwRDqn" => :haystack})
    (phoenix_live_view 0.15.0-dev) lib/phoenix_live_view/upload.ex:167: Phoenix.LiveView.Upload.get_upload_by_ref!/2
    (phoenix_live_view 0.15.0-dev) lib/phoenix_live_view/upload.ex:107: Phoenix.LiveView.Upload.update_progress/4
    (phoenix_live_view 0.15.0-dev) lib/phoenix_live_view/channel.ex:133: Phoenix.LiveView.Channel.handle_info/2
    (stdlib 3.13.2) gen_server.erl:680: :gen_server.try_dispatch/4
    (stdlib 3.13.2) gen_server.erl:756: :gen_server.handle_msg/6
    (stdlib 3.13.2) proc_lib.erl:226: :proc_lib.init_p_do_apply/3
Last message: %Phoenix.Socket.Message{event: "progress", join_ref: "12", payload: %{"entry_ref" => "0", "event" => nil, "progress" => %{"error" => "failed"}, "ref" => "phx-FkXg9nGnf0i-hDqH"}, ref: "15", topic: "lv:phx-FkXg9mbltojoQDpH"}
State: %{components: {%{}, %{}, 1}, join_ref: "12", serializer: Phoenix.Socket.V2.JSONSerializer, socket: #Phoenix.LiveView.Socket<assigns: %{flash: %{}, intervals: nil, live_action: :index, min_length: nil, uploaded_files: [], uploads: %{__phoenix_refs_to_names__: %{...}, ...}}, changed: %{}, endpoint: GuideSearchUiWeb.Endpoint, id: "phx-FkXg9mbltojoQDpH", parent_pid: nil, root_pid: #PID<0.1657.0>, router: GuideSearchUiWeb.Router, view: GuideSearchUiWeb.PageLive, ...>, topic: "lv:phx-FkXg9mbltojoQDpH", transport_pid: #PID<0.1645.0>}
```

**Comment by Manwholikespie** [comment]
Same here. The validate and newer Phoenix fixed it, I'm guessing 1.5.3 needs https://github.com/phoenixframework/phoenix/commit/cf098e9cf7a44ee6479d31d911a97d3c7430c6fe
